### ISSUE TITLE
Transfer Process never reaches final state

### ISSUE THREAD

**Issue by JulianLegler** [GAIA-X4PLC-AAD/EDC-Blockchain-Catalog#16]
# Bug Report

## Describe the Bug
When starting a HttpData transfer process, the process will start and also send the data successful. After that, the state will not proceed to a finished state. 

### Expected Behavior
When the data is send, the state should be shown as finished

### Observed Behavior
State appears to be unable to overcome STARTED status 

## Steps to Reproduce
1. Do all steps to have a approved contract negotiation.
2. start a HttpData transfer process
3. See that the data is received correctly
4. have a look at the transfer process status at the dashboard or via API
5. See that it will stay in "STARTED"

## Addition
I slightly remembered that at some point there was an error indicating that there is actually missing some component which is required to tell when a transfer is finished.

**Comment by JulianLegler** [comment]
Seem like it would require to implement the [`StatusChecker`](https://github.com/eclipse-edc/Connector/blob/v0.2.1/spi/control-plane/transfer-spi/src/main/java/org/eclipse/edc/connector/transfer/spi/types/StatusChecker.java) interface and register it with the [`StatusCheckerRegistry`](https://github.com/eclipse-edc/Connector/blob/v0.2.1/spi/control-plane/transfer-spi/src/main/java/org/eclipse/edc/connector/transfer/spi/status/StatusCheckerRegistry.java) for the data destination type "HttpData".

**Comment by robertschubert** [comment]
@JulianLegler the `StatusChecker` concept was removed with version 0.3.1

Issue: https://github.com/eclipse-edc/Connector/issues/3448
PR: https://github.com/eclipse-edc/Connector/pull/3480
Changelog: https://github.com/eclipse-edc/Connector/releases/tag/v0.3.1

There are several errors in the log both after contract negotiation and data transfer. The errors say basically that some json could not be parsed. I guess this is the reason why. I guess if we would update to a newer version the error is gone. 
There may be a risk that we break our Dashboard then in case of breaking changes have affect on the frontend.

I already updated to 0.3.1 in a branch, PR is upcoming.
